src/main/kotlin/lotto/domain/LottoPrize.kt

```diff
@@ -0,0 +1,32 @@
+package lotto.domain
+
+import lotto.domain.enums.PrizeType
+import lotto.domain.lotto.Lotto
+import lotto.domain.lotto.WinningLotto
+
+class LottoPrize(
+    userLottoList: List<Lotto>,
+    private val winningLotto: WinningLotto
+) {
+
+    private val _prizeResult: MutableMap<Int, Int> = (3..6)
+        .associateWith { 0 } as MutableMap<Int, Int>
+
+    val prizeResult: Map<Int, Int> get() = _prizeResult
+
+    init {
+        userLottoList
+            .forEach { recordLottoPrize(it) }
+    }
+
+    fun totalPrizeMoney(): Int = _prizeResult
+        .filter { it.value != 0 }
+        .map { PrizeType.findPrizeMoney(it.key) }
+        .sum()
+
+    private fun recordLottoPrize(lotto: Lotto) {
+        val answer = lotto.lottoNumber.filter { winningLotto.containsLottoNumber(it) }.size
```

Comment:
```suggestion
        val answer = lotto
            .lottoNumber
            .filter { winningLotto.containsLottoNumber(it) }
            .size
```
함수 체이닝 컨벤션을 적용하면 좋을 것 같아요

Reply:
8757f11 일급 컬렉션을 적용해 보았습니다 그리고 init 부분을 따로 LottoMatch를 시켜주는 클래스로 분할해보았습니다.